File: GuardianMonsters/core/src/de/limbusdev/guardianmonsters/fwmengine/world/ui/EntitySprite.java
package de.limbusdev.guardianmonsters.fwmengine.world.ui;

import com.badlogic.gdx.graphics.g2d.Sprite;
import com.badlogic.gdx.graphics.g2d.TextureRegion;

/**
 * Extends {@link Sprite} adding a boolean for switching sprites visibility. This will cause
 * {@link ExtendedTiledMapRenderer} to not draw an
 * EntitySprite if its visibility is set to false.
 * Created by georg on 21.11.15.
 */
public class EntitySprite extends Sprite {
    /* ............................................................................ ATTRIBUTES .. */
    public boolean visible;
    /* ........................................................................... CONSTRUCTOR .. */
    public EntitySprite(TextureRegion textureRegion) {
        super(textureRegion);
        this.visible = true;
    }
    /* ............................................................................... METHODS .. */
    
    /* ..................................................................... GETTERS & SETTERS .. */
}

File: GuardianMonsters/core/src/de/limbusdev/guardianmonsters/fwmengine/world/ui/EntitySprite.kt
package de.limbusdev.guardianmonsters.fwmengine.world.ui

import com.badlogic.gdx.graphics.g2d.Sprite
import com.badlogic.gdx.graphics.g2d.TextureRegion

/**
 * Extends [Sprite] adding a boolean for switching sprites visibility. This will cause
 * [ExtendedTiledMapRenderer] to not draw an
 * EntitySprite if its visibility is set to false.
 *
 * @author Georg Eckert 2015-11-21
 */
class EntitySprite(textureRegion: TextureRegion) : Sprite(textureRegion)
{
    // --------------------------------------------------------------------------------------------- PROPERTIES
    var visible: Boolean = false


    // --------------------------------------------------------------------------------------------- CONSTRUCTORS
    init
    {
        visible = true
    }
}
